config.splitting.js

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright 2019 Algolia Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+// This config can split long content into more than 1 record per resource.
```

Comment:
I think more doc is needed:
- What is the splitting strategy (explain that we will split on TEXT_ELEMENTS first and then on length)
- What are the constants to change depending on my documents/use case

Reply:
> What are the constants to change depending on my documents/use case

I'm not sure what I can reply to this... Don't hesitate to clarify what you meant, if you think that I should change something.

That said, I did add more explanations about the splitting strategy.

Reply:
I think it's ok with the new explanation.
What I meant was: if I have a pdf where pages are identified by `<div class="page">`, if I want to split by pages I will set `const TEXT_ELEMENTS = ['.page'];`

Reply:
Thanks for having raised that issue!
Indeed, I was a bit biased by DocSearch's implementation: create one record per paragraph instead of fetching the whole page (i.e. `<body>`) and splitting from there. 
While it's good to give some flexibility to config authors, it does make more sense to not split per paragraph by default, notably because doing this results in generating many more records.
=> Here's what I did:
- I set the default `TEXT_ELEMENTS = ['body'];` (instead of `['p', 'li']`) and added more explanations about how this value impacts the splitting logic;
- this surfaced two problems that I fixed:
  - I made the `turnEmptyTagsToSeparators()` function recursive, to also convert tags found at deeper levels of the DOM;
  - I also made the splitting logic work on JSON-serialized strings instead of pure strings, because the max record size was exceeded when special characters (e.g. `\"`) were found in the content.
Wdyt?